File: lib/models/question.dart
class Question {
  final String id;
  final String studentName;
  final String studentAvatar;
  final String courseTitle;
  final String question;
  final DateTime timestamp;
  final bool answered;
  final String? answer;

  Question({
    required this.id,
    required this.studentName,
    required this.studentAvatar,
    required this.courseTitle,
    required this.question,
    required this.timestamp,
    required this.answered,
    this.answer,
  });

  // Convert to JSON for API communication
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'studentName': studentName,
      'studentAvatar': studentAvatar,
      'courseTitle': courseTitle,
      'question': question,
      'timestamp': timestamp.toIso8601String(),
      'answered': answered,
      'answer': answer,
    };
  }

  // Create from JSON response
  factory Question.fromJson(Map<String, dynamic> json) {
    return Question(
      id: json['_id'] ?? json['id'] ?? '',
      studentName: json['studentName'] ?? '',
      studentAvatar: json['studentAvatar'] ?? 'assets/images/default_avatar.jpg',
      courseTitle: json['courseTitle'] ?? '',
      question: json['text'] ?? json['question'] ?? '',
      timestamp: json['timestamp'] != null ? 
          DateTime.parse(json['timestamp']) : 
          DateTime.now(),
      answered: json['answered'] ?? false,
      answer: json['answer'],
    );
  }
}

File: edubridge-backend/models/Question.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const QuestionSchema = new Schema({
  studentName: {
    type: String,
    required: true
  },
  studentId: {
    type: Schema.Types.ObjectId,
    ref: 'users',
    required: true
  },
  studentAvatar: {
    type: String,
    default: 'assets/images/default_avatar.jpg'
  },
  courseTitle: {
    type: String,
    required: true
  },
  courseId: {
    type: Schema.Types.ObjectId,
    ref: 'courses'
  },
  text: {
    type: String,
    required: true
  },
  timestamp: {
    type: Date,
    default: Date.now
  },
  answered: {
    type: Boolean,
    default: false
  },
  answer: {
    type: String,
    default: null
  },
  teacherId: {
    type: Schema.Types.ObjectId,
    ref: 'users'
  }
});

module.exports = Question = mongoose.model('question', QuestionSchema);
